sympy/series/limits.py

```diff
@@ -283,6 +324,8 @@ def set_signs(expr):
                 if ex > 0:
                     return S.Zero
                 elif ex == 0:
+                    if z0.is_Symbol and not z0 in e.free_symbols:
+                        coeff = return_piecewise(coeff, z0, cdir)
```

Comment:
`return return_piecewise(...`

(which probably shows that the name is actually not that good, maybe `_limit_for_symbolic_point` or something similar?)

Reply:
Yeah makes much more sense than what something like `return_piecewise` would convey . The name suggested justifies the work correctly !